### PR TITLE
Created count_of_trailing_ciphers_in_factorial_n.cpp

### PR DESCRIPTION
**Pull request by Swastyy**
#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes: 

<a href="https://gitpod.io/#https://github.com/TheAlgorithms/C-Plus-Plus/pull/1512"></a>

### CODE REVIEW

**Comment by Swastyy** [comment]
@ayaankhan98 @Panquesito7 Please check it out once.

**Review by Panquesito7 — CHANGES_REQUESTED** [review]
Please fix [`clang-tidy`](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1512/checks?check_run_id=2888348032) warnings.

**Comment by Swastyy** [comment]
> Please fix [`clang-tidy`](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1512/checks?check_run_id=2888348032) warnings.

Done!!
Fixed the clang-tidy warnings.

File: math/count_of_trailing_ciphers_in_factorial_n.cpp [hunk 1]
```diff
@@ -0,0 +1,64 @@
+/**
+ * @file
+ * @brief A simple program to count the number of ciphers in a factorial of some
```
Comment (ayaankhan98) [inline math/count_of_trailing_ciphers_in_factorial_n.cpp:3]:
```suggestion
 * @brief Program to count number of ciphers in `n!`
```

File: math/count_of_trailing_ciphers_in_factorial_n.cpp [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+ * number.
```
Comment (ayaankhan98) [inline math/count_of_trailing_ciphers_in_factorial_n.cpp:4]:
```suggestion
```

File: math/count_of_trailing_ciphers_in_factorial_n.cpp [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+ * @author [Swastika Gupta](https://github.com/swastyy)
+ */
+#include <cassert>
+#include <iostream>
+
+/**
+ * Function to count the number of ciphers.
```
Comment (ayaankhan98) [inline math/count_of_trailing_ciphers_in_factorial_n.cpp:11]:
```suggestion
 * @brief Function count and returns number of ciphers in `n!`
```

File: math/count_of_trailing_ciphers_in_factorial_n.cpp [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+ * @param n is the number to be given as argument.
```
Comment (ayaankhan98) [inline math/count_of_trailing_ciphers_in_factorial_n.cpp:12]:
```suggestion
 * @param n Number for which `n!` ciphers are returned
```

File: math/count_of_trailing_ciphers_in_factorial_n.cpp [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+ * @return it returns the power of 5 in factorial of a given number.
```
Comment (ayaankhan98) [inline math/count_of_trailing_ciphers_in_factorial_n.cpp:13]:
```suggestion
 * @return count Number of ciphers in `n!`.
```

File: math/count_of_trailing_ciphers_in_factorial_n.cpp [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    if (count_of_number_of_ciphers_in_factorial_n(871) == 215) {
+        std::cout << "passed\n";
+    }
+    std::cout << "failed\n";
+
+    std::cout << "Test 4:\t n=239\n";
+    if (count_of_number_of_ciphers_in_factorial_n(239) == 57) {
+        std::cout << "passed\n";
+    }
+    std::cout << "failed\n";
```
Comment (ayaankhan98) [inline math/count_of_trailing_ciphers_in_factorial_n.cpp:55]: whether the tests passes or not, in both cases `Failed` will be displayed, as it is priting unconditionally.

Comment (ayaankhan98) [reply math/count_of_trailing_ciphers_in_factorial_n.cpp:54]: It shoudl be like
```C++
void test() {
    cout << "Running predefined tests...\n";
    assert(count_of_number_of_ciphers_in_factorial_n(871) == 215);
    assert(count_of_number_of_ciphers_in_factorial_n(977) == 242);
...............
............... 
..............// write some more tests covering corner cases
................
...............
    std::cout << "Test Passed...!\n";
}
```

Comment (Swastyy) [reply math/count_of_trailing_ciphers_in_factorial_n.cpp:54]: Initially I wrote the "failed" thing inside the else block but it was showing warnings then I searched over internet on how to remove these warnings. I saw this article https://clang.llvm.org/extra/clang-tidy/checks/readability-else-after-return.html where its removing else to remove the warnings.

Comment (Swastyy) [reply math/count_of_trailing_ciphers_in_factorial_n.cpp:54]: Ohh Ohkk! Got it!

Comment (ayaankhan98) [reply math/count_of_trailing_ciphers_in_factorial_n.cpp:54]: observe the pattern of execution of what you have written. See if the first test cases passes then program control will go inside the first `if` block and there it prints `passed` and after printing this, program control will come out of the first `if` block and it again gets a print command which will print `failed`, Meaning for a single test it will print both `passed` as well  as `failed` which does not makes any sense. Hope you got it.

Comment (Swastyy) [reply math/count_of_trailing_ciphers_in_factorial_n.cpp:54]: Yep, thanks for pointing out.

Comment (Swastyy) [reply math/count_of_trailing_ciphers_in_factorial_n.cpp]: @ayaankhan98 I have done the changes. Pl have a look

**Comment by Swastyy** [comment]
@Panquesito7 this pull request has a label as "automated test are failing" I want to know which test are failing like if there is any need to change the data types of variables (like to long long etc).

**Review by Panquesito7 — COMMENTED** [review]
> I want to know which test are failing

The [CodeQL](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1512/checks?check_run_id=2911353484) tests are failing due to an error in your code.
By the way: I cannot review properly your PR if your fork is deleted.

**Comment by Swastyy** [comment]
> > I want to know which test are failing
> 
> The [CodeQL](https://github.com/TheAlgorithms/C-Plus-Plus/pull/1512/checks?check_run_id=2911353484) tests are failing due to an error in your code.
> By the way: I cannot review properly your PR if your fork is deleted.

@Panquesito7 Yea, accidently it got deleted. Then I forked it again. Will it create any problem in this pull request or is it ok? (so that I can check the errors from here only)

**Comment by Panquesito7** [comment]
> Will it create any problem in this pull request?

No, it does not create any problem, but I can't make suggestions easily due to some problem.
It'd be great if you could re-create this pull request for it to be reviewed more easily.

Thanks. 🙂

**Comment by Swastyy** [comment]
> > Will it create any problem in this pull request?
> 
> No, it does not create any problem, but I can't make suggestions easily due to some problem.
> It'd be great if you could re-create this pull request for it to be reviewed more easily.
> 
> Thanks. 🙂

Yeah ok, I will recreate this PR but please see this PR (https://github.com/TheAlgorithms/C-Plus-Plus/pull/1511) it has been approved, will you merge it from there or I have to recreate it too(since it was with the same fork that got deleted)? @Panquesito7

**Comment by Panquesito7** [comment]
You can re-create both PRs, and I'll review them when I can; that'd be great, thanks. 🙂

**Comment by Swastyy** [comment]
> You can re-create both PRs, and I'll review them when I can; that'd be great, thanks. 🙂

Here they are Trailing ciphers count in n! (https://github.com/TheAlgorithms/C-Plus-Plus/pull/1521) and N-bonacci series (https://github.com/TheAlgorithms/C-Plus-Plus/pull/1518)

**Comment by Panquesito7** [comment]
Thank you. I'll close this PR in favor of the other ones.
